### PR TITLE
[RI2] Converting docs from markdown to rst

### PR DESCRIPTION
**Pull request by gkunz**
This PR implements cleanup of the documentation and conversion from markdown to rst.

Implements #2649

### CODE REVIEW

**Comment by tomkivlin** [comment]
Just as an FYI as I don't want to undo any good work already done, but internally I have moved from using recommonmark as the markdown parser to myst, which allows native sphinx directives and roles within a markdown document, but the rest of the doc remains in markdown.

e.g. for images: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-images

(it also deals with copying images and other artefacts automatically, so no need for any "direct_copy" script any more.

**Comment by CsatariGergely** [comment]
> Regarding the branch, I agree that it is helpful to keep things clean this way. However, I "noticed" that Sphinx happily works with both formats in parallel - as for instance demonstrated by this PR which creates rst files among markdown files. So, we could actually convert chapters one-by-one in the same tree - thereby also minimizing the problem of diverging content between master and the integration branch. On the other hand, we'd need a commitment by everyone to go ahead with rst to avoid mixed formats. Yet, even that wouldn't be a problem, i.e., leave it up to the WSLs to decide the format they prefer.

Now I agree that this approach without the integration branch is better, so let's follow this approach. I will do the same with my pr-s.

Yes, in this way we can leave it to the sub-projects to decide on what tooling they need. On the other side I would gradually stop using recommonmark as it is not supported and forces us to use an old version of Sphinx.

**Comment by CsatariGergely** [comment]
> Just as an FYI as I don't want to undo any good work already done, but internally I have moved from using recommonmark as the markdown parser to myst, which allows native sphinx directives and roles within a markdown document, but the rest of the doc remains in markdown.
> 
> e.g. for images: https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#syntax-images
> 
> (it also deals with copying images and other artefacts automatically, so no need for any "direct_copy" script any more.

Looks like a good option, but for the sub-projects handed over to GSMA rst is the desired format. For other sub-projects this could be a viable option if GitHub is able to render it correctly.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:27]:
```suggestion
Terminology in this document will follow `Anuket Terminology <../../../common/glossary.md>`__.
```

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:71]: What about the Figure caption?

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:10]: Can we define these references at the end like the figures? Easy to change if a reference link changes or there is an error, etc. Thanks.

**Comment by collivier** [comment]
Agree @pgoyal01 . please add linkcheck (+docutils) which will automatically detect all false links to md.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter01.rst:10]: yes. rst typically makes use of named references, i.e., there is a global name defined at the target and that is being referred to from everywhere. However, we should introduce this over time as it requires the target anchors to exist.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter01.rst]: added in the latest update. I am also using the rst reference mechanism to reference the figure (by name).

**Comment by gkunz** [comment]
@collivier added doc8. No link checking since I don't think we should use absolute links.

**Comment by gkunz** [comment]
Now that the other chapters are getting merged as rst, it would be good to get this reviewed and merged as well. Thank you!

File: doc/ref_impl/cntt-ri2/README.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/README.rst:4]:
```suggestion
This is the Kubernetes-based Reference Implementation (RI-2).
```

File: doc/ref_impl/cntt-ri2/README.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/README.rst:25]: do we need to a Lakelse release info

Comment (gkunz) [reply doc/ref_impl/cntt-ri2/README.rst:25]: yes, that's the problem with keeping these things open for a long time. I'll merge the latest updates. Thanks for the catching this.

File: doc/ref_impl/cntt-ri2/chapters/chapter01.rst
Comment (pgoyal01) [inline doc/ref_impl/cntt-ri2/chapters/chapter01.rst:74]: what about the figure caption?

File: doc/ref_impl/cntt-ri2/README.rst
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/README.rst:26]: included in https://github.com/cntt-n/CNTT/pull/2656/commits/e781dee7fd59ba28d36593c292b504b9b2e9d529

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by gkunz** [comment]
just realized that this is broken - DO NOT MERGE

**Comment by pgoyal01** [comment]
@gkunz I added a label "WIP" and recycled the approvals -- would need re-approvals from @rihabbanday and me.

**Comment by gkunz** [comment]
Fixed, sorry for the hick-up.

File: doc/ref_impl/cntt-ri2/chapters/chapter04.rst
Comment (collivier) [inline doc/ref_impl/cntt-ri2/chapters/chapter04.rst:14]: this won't work anymore. You could simply run linkcheck to detect all of them.

**Review by collivier — CHANGES_REQUESTED** [review]

File: doc/ref_impl/cntt-ri2/chapters/chapter04.rst
Comment (gkunz) [reply doc/ref_impl/cntt-ri2/chapters/chapter04.rst]: @collivier: fixed. the last commit switched to references where possible and passed linkcheck.

**Comment by gkunz** [comment]
@collivier I addressed your last comment. Could you please re-review?

**Review by pgoyal01 — APPROVED** [review]

**Review by rihabbanday — APPROVED** [review]

**Comment by gkunz** [comment]
HI @collivier: I addressed your comments. Can you please re-review and/or approve? Thanks

**Review by michaelspedersen — APPROVED** [review]
Went over the changes, focusing on the content (text, references). I did not find any immediate errors compared to the previous content.

I have not tested the build process and looked at the final output.

**Comment by collivier** [comment]
@gkunz I'm fine de merge it. I don't see why I'm not allowed to write comments  in it.
I would suggest to wrap lines to 80 via pandoc, rst -> rst (it will also mange very long url).
I think sphinx config could be simplified.

I see minor edit changes in the PR. Would it make sens to fix any typo in the latest stable branch too?

Thank you to add the linkcheck part.

**Review by collivier — APPROVED** [review]

**Pull request merged**
